docs/docs/add-seo-component.md

```diff
@@ -0,0 +1,199 @@
+---
+title: "Adding a SEO component"
+---
+
+Every website on the web has basic _meta-tags_ like the title, favicon or description of the page in their `<head>` element. This information gets displayed in the browser and is used when someone, e.g. shares your website on Twitter. You can give your users and these websites additional data to embed your website with more data — and that's where this guide for a SEO component comes in. At the end you'll have a component you can place in your layout file and
+have rich previews for other clients, smartphone users, and search engines.
+
+_Note: This component will use StaticQuery. If you're unfamiliar with that, have a look at the [StaticQuery documentation](/docs/static-query/). You also have to have `react-helmet` installed for which you can have a look at [this document](/docs/add-page-metadata)._
+
+## gatsby-config.js
+
+Gatsby makes all data put into the `siteMetadata` section of your `gatsby-config` file automatically available in GraphQL and therefore it's a good idea to place your information for the component there.
+
+```js:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    title: 'Severus Snape',
+    titleTemplate: '%s · The Real Hero',
+    description: 'Hogwarts Potions master, Head of Slytherin house and former Death Eater.',
+    url: 'https://www.doe.com', // No trailing slash allowed!
+    image: '/images/snape.jpg', // Path to your image you placed in the 'static' folder
+    twitterUsername: '@occlumency',
+  },
+}
+```
+
+## SEO component
+
+Create a new component with this initial boilerplate:
+
+```jsx:title=src/components/SEO.js
+import React from "react"
+import Helmet from "react-helmet"
+import PropTypes from "prop-types"
+import { StaticQuery } from "gatsby"
+
+const SEO = ({ title, description, image, pathname, article }) => ()
+
+export default SEO
+
+SEO.propTypes = {
+  title: PropTypes.string,
+  description: PropTypes.string,
+  image: PropTypes.string,
+  pathname: PropTypes.string,
+  article: PropTypes.bool,
+}
+
+SEO.defaultProps = {
```

Comment:
I kinda like the idea of _not_ including these. Not a strong reservation though!

Reply:
Or at least not including the ones that are null!

Reply:
Why so?

Reply:
undefined will be handled the same way as null, i.e. falsy.

What's the benefit? I can understand defaulting something to an _actual_ value, but not sure I get setting to null!

Reply:
I don't have strong feelings about whether using them or not, so whatever the majority thinks. I *personally* use it this way.

Reply:
@dschau Ah, didn’t think of, but my eslint will complain if I don’t set defaultProps when the other props are not `.isRequired`, so technically this would be the more correct way I guess